client.js

```diff
@@ -742,11 +742,14 @@ function createGetResponse (id, token, value) {
   return r
 }
 
-function encodePeer (host, port) {
-  const buf = Buffer.allocUnsafe(6)
-  const ip = host.split('.')
-  for (let i = 0; i < 4; i++) buf[i] = parseInt(ip[i] || 0, 10)
+function encodePeer(host, port) {
+  const buf = Buffer.alloc(6)
```

Comment:
This does have a minor logic change in that this will initialize a 0 filled Buffer. I do not know enough about this project to say if that matters.